### DISCUSSION TITLE
Video Guide

### DISCUSSION THREAD

**Discussion by IamMEKing** [DIGITALCRIMINALS/OnlyFans#106] [Q&A]
Can anyone make a video guide on how to do this for first time users?

**Comment by stranger-danger-zamu** [comment]
Git Gud. \s


If you are too stupid or lazy to read, then pray someone else takes these instructions and makes a video from them. I _**won't**_ be helping people individually.

That said, this is just very verbose for the technically inept.

---
There really isn't much to do that a video will be able to help with. 

Assuming you are comfortable using the command line and already downloaded the compatible Python runtime, the basic steps are:

1. Grab a copy of the code, preferably the latest released version. The command below should do that automatically. Using the "master" branch can lead to problems.
    ```bash
    # determine which is the newest released version
    export LATEST_RELEASE_TAG=$(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://github.com/DIGITALCRIMINALS/OnlyFans.git|tail -n1 |grep -oP 'v[\d.]+$')
    # grab the version from git, but only the single branch without any history
    git clone -b $LATEST_RELEASE_TAG --depth 1 https://github.com/DIGITALCRIMINALS/OnlyFans.git
    # move into the folder
    cd OnlyFans
    ```
2. Install dependencies to be able to run the program. There are a bunch of ways to do this though and some of them are better than others. Here is the "best practice" way to do this.
    ```bash
    # make a "virtualenv" which is nerd talk for a folder which 
    # will house all the dependencies
    python3 -m venv --copies venv
    # instead of activating the virtualenv (which is nerd talk for 
    # setting the terminal to use it over the system installation 
    # of python), we will just use it manually
    ./venv/bin/python3 -m pip install -r requirements.txt
    ```
3. Now that we have the dependencies met, lets run this so it sets up the configuration folders we have to edit.
    ```bash
    # we have to use our virtualenv python to run this or else 
    # our dependencies will be missing!
    ./venv/bin/python3 start_ofd.py
    ```
4. It should print a bunch of stuff and tell you to go edit files. IIRC, it should just exit by hitting enter a few times and complaining a bunch, so let's do that.
5. Now, before we go get our info from the browser, lets quickly talk about the `.profile`,`.sites`,and `.settings` folders and their configuration files.
    > * `.profile` houses configuration files where our login info goes. Each site has it's own internal directory and each site supports more than one login. For now we'll assume you are just using this for OF. So we can just look at `.profiles/OnlyFans/default/auth.json`.
    > * `.settings` houses the configuration for how the program actually does it stuff. Hate how it asks which models and then which things to capture? This is where you can change all that. It is documented pretty well in the README but for now we will leave this alone since we want to change the least amount of stuff.
    > * `.sites` houses where all the files are saved to by default. There is nothing to modify here, literally just folders and the files it downloads.
6. Okay with that info, we know that we'll have to modify the `.profile/OnlyFans/default/auth.json` to login and grab all the stuff.
    
    * This is close to what the file looks like. We just need to replace the `<COOKIE>`, `<XBC>`, and `<USERAGENT>` with the values from what we get from the browser.
    ```json
    {
        "auth": {
            "username": "default",
            "cookie": "<COOKIE>",
            "x_bc": "<XBC>",
            "user_agent": "<USERAGENT>",
            "email": "",
            "password": "",
            "hashed": false,
            "support_2fa": false,
            "active": true
        }
    }
    ```

    * I think that the README is pretty good at explaining getting the values from your browser. But just incase you can't find the new values, here is a step by step guide.

        1. We need to be sure that there aren't any conflicting cookies or other bits of data, so we need a clean browser. An easy way to do this is with your browser's Private Window/Incognito Window. However, they are finky, so we need to close down all the browser windows once and *then* open a private tab. So do that.
        2. Now, we go to `Onlyfans.com` in your private window, but don't login yet. First open the `Developer Tools` in your browser. `F12` is pretty consistently the way to open it (you might need to use the function key if it's also mapped to something like volume control).
        3. Now in the Developer Pane go to the `Network` tab. The tabs should be along the top of the pane (you might need to scroll or click on the `>>` to show all the tabs). 
        4. In the `Network` tab, look for a box that says `Filter` or `Filter URLs`. Type in `init`.
        5. Now go back to the `onlyfans` pane and login. In the Developer window you'll get a few rows with `init` being the top most. Click on it until a side bar opens up to the right.
        6. In the sidebar now scroll to the bottom and there you'll see `x-bc` and `user-agent`. 
            * Right click x-bc and select `Copy Value`. Replace `<XBC>` in `Onlyfans/default/auth.json` with this value (you can just paste it in).
            * Right click user-agent, select `Copy Value`. Replace `<USERAGENT>` with this value (you can just paste it in).
        7. Finally we just to copy the cookies, so scroll up and you'll find `cookie:`. Similar to the other two, just right click, `Copy Value`, and replace the `<COOKIES>` in `Onlyfans/default/auth.json` file.
        8. Save the auth file.
7. Now run it.
    ```bash
    ./venv/bin/python start_ofd.py
    ```
    Now you can just enter 0 for nearly all the values and it'll just download everything. Careful though since some models have hundreds of gigabytes of content (I know of one cam model who just upload the shows in 4k to their OF, downloaded at least 240GBs before I noticed 🤮).

**Reply by IamMEKing** [reply]
./venv/bin/python3 start_ofd.py

:~/OnlyFans$ ./venv/bin/python3 start_ofd.py
Traceback (most recent call last):
  File "start_ofd.py", line 15, in <module>
    import tests.main_test as main_test
  File "/home/rush/OnlyFans/tests/main_test.py", line 58
    match string_match:
          ^
SyntaxError: invalid syntax

This is what pops up after that

**Reply by stranger-danger-zamu** [reply]
Install python 3.10

**Reply by IamMEKing** [reply]
installed and ive made sure its the correct version being used
